cmd/connectivity_pod_to_pod.go

```diff
@@ -24,73 +19,42 @@ Example:
 	(add example)
 `
 
-type connectivityPodToPodCmd struct {
-	out             io.Writer
-	fromPod         string
-	toPod           string
-	clientSet       kubernetes.Interface
-	smiAccessClient smiAccessClient.Interface
-	// meshConfigClient osmConfigClient.Interface
-	restConfig *rest.Config
-}
-
-func newConnectivityPodToPodCmd(config *action.Configuration, in io.Reader, out io.Writer) *cobra.Command {
-	podToPodCmd := &connectivityPodToPodCmd{
-		out: out,
-	}
-
+func newConnectivityPodToPodCmd() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "pod-to-pod SOURCE_POD DESTINATION_POD",
 		Short: "Checks connectivity between Kubernetes pods",
 		Long:  connectivityPodToPodDesc,
 		Args:  cobra.ExactArgs(2),
 		RunE: func(_ *cobra.Command, args []string) error {
-			podToPodCmd.fromPod = args[0]
-			podToPodCmd.toPod = args[1]
+			if len(args) < 2 {
+				return errors.Errorf("provide both SOURCE_POD and DESTINATION_POD")
+			}
 
 			config, err := settings.RESTClientGetter().ToRESTConfig()
 			if err != nil {
 				return errors.Errorf("Error fetching kubeconfig: %s", err)
 			}
 
-			podToPodCmd.restConfig = config
-
 			clientset, err := kubernetes.NewForConfig(config)
 			if err != nil {
 				return errors.Errorf("Could not access Kubernetes cluster, check kubeconfig: %s", err)
 			}
-			podToPodCmd.clientSet = clientset
+			clientSet := clientset
 
-			accessClient, err := smiAccessClient.NewForConfig(config)
+			fromPod, err := kuberneteshelper.PodFromString(args[0])
 			if err != nil {
-				return errors.Errorf("Could not initialize SMI Access client: %s", err)
+				return errors.New("invalid SOURCE_POD")
 			}
-			podToPodCmd.smiAccessClient = accessClient
 
-			// configClient, err := osmConfigClient.NewForConfig(config)
-			// if err != nil {
-			// 	return errors.Errorf("Could not initialize OSM Config client: %s", err)
-			// }
-			// podToPodCmd.meshConfigClient = configClient
+			toPod, err := kuberneteshelper.PodFromString(args[1])
+			if err != nil {
+				return errors.New("invalid DESTINATION_POD")
+			}
 
-			return podToPodCmd.run()
+			connectivity.PodToPod(clientSet, fromPod, toPod)
+			return nil
 		},
 		Example: connectivityPodToPodExample,
 	}
 	return cmd
 }
-
-func (podToPodCmd *connectivityPodToPodCmd) run() error {
```

Comment:
I'm curious about what prompted this change? It seems like most - if not all - osm cli commands follow this format of doing the "work" in the `run` function of the command

Reply:
I had observed a few things:
1. We tend to create structs (which then gets a `run()`). These structs have fields, which are unused. Also the actual structs seem to be unused. But the fact that we assign a value to a field of a struct, masks an issue of these variables being unused. So both the struct and the assignments are unused. Here is an example: https://github.com/openservicemesh/osm-health/blob/def72f6b3b97ecbefde9d24a2f9b50034e64a6f1/cmd/connectivity_pod_to_pod.go#L27-L35
2. We do a bunch of work before we call `run()`, then we call run and do some more work. But the actual `run()` is actually an invocation to some other function. I see this as an opportunity to remove complexity: https://github.com/openservicemesh/osm-health/blob/def72f6b3b97ecbefde9d24a2f9b50034e64a6f1/cmd/connectivity_pod_to_pod.go#L48-L75

Reply:
Got it, makes sense